// @mongez/fs/cjs/empty.js
'use strict';var getFile=require('./getFile.js'),list=require('./list.js');/**
 * Check if the given file path is empty
 *
 * If file does not exist, it will return true
 */
function isEmptyFile(path) {
    try {
        return getFile.getFile(path).length === 0;
    }
    catch (error) {
        return true;
    }
}
/**
 * Check if the given file path is not empty
 */
function isNotEmptyFile(path) {
    return !isEmptyFile(path);
}
/**
 * Check if the given directory path is empty
 */
function isEmptyDirectory(path) {
    try {
        return list.list(path).length === 0;
    }
    catch (error) {
        return true;
    }
}
/**
 * Check if the given directory is not empty
 */
function isNotEmptyDirectory(path) {
    return !isEmptyDirectory(path);
}
// Async version
async function isEmptyFileAsync(path) {
    return new Promise(async (resolve) => {
        try {
            resolve((await getFile.getFileAsync(path)).length === 0);
        }
        catch (error) {
            resolve(true);
        }
    });
}
async function isNotEmptyFileAsync(path) {
    return !(await isEmptyFileAsync(path));
}
async function isEmptyDirectoryAsync(path) {
    return new Promise(async (resolve) => {
        try {
            resolve((await list.listAsync(path)).length === 0);
        }
        catch (error) {
            resolve(true);
        }
    });
}
async function isNotEmptyDirectoryAsync(path) {
    return !(await isEmptyDirectoryAsync(path));
}exports.isEmptyDirectory=isEmptyDirectory;exports.isEmptyDirectoryAsync=isEmptyDirectoryAsync;exports.isEmptyFile=isEmptyFile;exports.isEmptyFileAsync=isEmptyFileAsync;exports.isNotEmptyDirectory=isNotEmptyDirectory;exports.isNotEmptyDirectoryAsync=isNotEmptyDirectoryAsync;exports.isNotEmptyFile=isNotEmptyFile;exports.isNotEmptyFileAsync=isNotEmptyFileAsync;//# sourceMappingURL=empty.js.map
