// @mongez/fs/cjs/getJsonFile.js
'use strict';var getFile=require('./getFile.js');/**
 * Load the given json file path
 *
 * @param path The path of the json file
 * @param reviver A function that transforms the results. This function is called for each member of the object.
 * If a member contains nested objects, the nested objects are transformed before the parent object is.
 */
function jsonFile(path, reviver) {
    try {
        const content = getFile.getFile(path);
        return JSON.parse(content, reviver);
    }
    catch (error) {
        throw error;
    }
}
const getJsonFile = jsonFile;
/**
 * Get async the given json file path
 */
async function jsonFileAsync(path, reviver) {
    try {
        const content = await getFile.getFileAsync(path);
        return JSON.parse(content, reviver);
    }
    catch (error) {
        throw error;
    }
}
const getJsonFileAsync = jsonFileAsync;exports.getJsonFile=getJsonFile;exports.getJsonFileAsync=getJsonFileAsync;exports.jsonFile=jsonFile;exports.jsonFileAsync=jsonFileAsync;//# sourceMappingURL=getJsonFile.js.map
