// @mongez/fs/cjs/lastModified.js
'use strict';var stats=require('./stats.js');/**
 * Get last modified time of the given path
 */
function lastModified(path) {
    try {
        return stats.stats(path).mtime;
    }
    catch (e) {
        return new Date();
    }
}
const fileLastModified = lastModified;
const directoryLastModified = lastModified;
/**
 * Get last modified time of the given path async
 */
async function lastModifiedAsync(path) {
    return new Promise(async (resolve, reject) => {
        try {
            resolve((await stats.statsAsync(path)).mtime);
        }
        catch (e) {
            reject(e);
        }
    });
}
const fileLastModifiedAsync = lastModifiedAsync;
const directoryLastModifiedAsync = lastModifiedAsync;exports.directoryLastModified=directoryLastModified;exports.directoryLastModifiedAsync=directoryLastModifiedAsync;exports.fileLastModified=fileLastModified;exports.fileLastModifiedAsync=fileLastModifiedAsync;exports.lastModified=lastModified;exports.lastModifiedAsync=lastModifiedAsync;//# sourceMappingURL=lastModified.js.map
