// @mongez/fs/cjs/makeDirectory.js
'use strict';var fs=require('fs'),exists=require('./exists.js');/**
 * Create new directory if the given path does not exist
 */
function makeDirectory(path, options = {
    mode: 0o777,
    recursive: true,
}) {
    return fs.mkdirSync(path, options);
}
const mkdir = makeDirectory;
const createDirectory = makeDirectory;
function ensureDirectory(path, options) {
    if (!exists.directoryExists(path)) {
        return makeDirectory(path, options);
    }
}
/**
 * Create new directory if the given path does not exist async
 */
async function makeDirectoryAsync(path, options = {
    mode: 0o777,
    recursive: true,
}) {
    return new Promise((resolve) => {
        fs.mkdir(path, options, () => {
            resolve();
        });
    });
}
const mkdirAsync = makeDirectoryAsync;
const createDirectoryAsync = makeDirectoryAsync;
async function ensureDirectoryAsync(path, options) {
    return new Promise(async (resolve) => {
        if (!(await exists.directoryExistsAsync(path))) {
            await makeDirectoryAsync(path, options);
        }
        resolve();
    });
}exports.createDirectory=createDirectory;exports.createDirectoryAsync=createDirectoryAsync;exports.ensureDirectory=ensureDirectory;exports.ensureDirectoryAsync=ensureDirectoryAsync;exports.makeDirectory=makeDirectory;exports.makeDirectoryAsync=makeDirectoryAsync;exports.mkdir=mkdir;exports.mkdirAsync=mkdirAsync;//# sourceMappingURL=makeDirectory.js.map
